GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -425,8 +425,8 @@ private void GetBranchRevisionFilter(ArgumentBuilder filter)
             }
             else if (IsShowCurrentBranchOnlyChecked || (IsShowFilteredBranchesChecked && !string.IsNullOrWhiteSpace(BranchFilter)))
             {
-                // See handling for --all below, explicitly add most recent stash
-                if (AppSettings.ShowStashes)
+                // See handling for --all below, explicitly add only most recent stash
+                if (AppSettings.ShowStashes && (!IsShowCurrentBranchOnlyChecked || !string.IsNullOrWhiteSpace(currentBranch.Value)))
```

Comment:
Reuse `AddFirstStashRef()`?

Reply:
I left a temp implementation, just for you to have something to comment on - or I forgot to remove it.